view/src/components/todo.js

```diff
@@ -2,25 +2,422 @@ import React, { Component } from "react";
 
 import withStyles from "@material-ui/core/styles/withStyles";
 import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import Dialog from "@material-ui/core/Dialog";
+import AddCircleIcon from "@material-ui/icons/AddCircle";
+import AppBar from "@material-ui/core/AppBar";
+import Toolbar from "@material-ui/core/Toolbar";
+import IconButton from "@material-ui/core/IconButton";
+import CloseIcon from "@material-ui/icons/Close";
+import Slide from "@material-ui/core/Slide";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import Card from "@material-ui/core/Card";
+import CardActions from "@material-ui/core/CardActions";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import CardContent from "@material-ui/core/CardContent";
+import MuiDialogTitle from "@material-ui/core/DialogTitle";
+import MuiDialogContent from "@material-ui/core/DialogContent";
+
+import axios from "axios";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+import { authMiddleWare } from "../util/auth";
 
 const styles = (theme) => ({
   content: {
     flexGrow: 1,
     padding: theme.spacing(3),
   },
+  appBar: {
+    position: "relative",
+  },
+  title: {
+    marginLeft: theme.spacing(2),
+    flex: 1,
+  },
+  submitButton: {
+    display: "block",
+    color: "white",
+    textAlign: "center",
+    position: "absolute",
+    top: 14,
+    right: 10,
+  },
+  floatingButton: {
+    position: "fixed",
+    bottom: 0,
+    right: 0,
+  },
+  form: {
+    width: "98%",
+    marginLeft: 13,
+    marginTop: theme.spacing(3),
+  },
   toolbar: theme.mixins.toolbar,
+  root: {
+    minWidth: 470,
+  },
+  bullet: {
+    display: "inline-block",
+    margin: "0 2px",
+    transform: "scale(0.8)",
+  },
+  pos: {
+    marginBottom: 12,
+  },
+  uiProgess: {
+    position: "fixed",
+    zIndex: "1000",
+    height: "31px",
+    width: "31px",
+    left: "50%",
+    top: "35%",
+  },
+  dialogeStyle: {
+    maxWidth: "50%",
+  },
+  viewRoot: {
+    margin: 0,
+    padding: theme.spacing(2),
+  },
+  closeButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: theme.palette.grey[500],
```

Comment:
Does the MUI theme implementation have semantic colors?  Like "secondary text color" or "icon foreground color"?  If so, it would be better to use those rather than hard-coding a color from the palette directly.

Reply:
the theme implementation does have semantic colors, yet I cannot move this exact grey into the semantic color definition because I can't call theme within its initialization. If you want I can find the hexadecimal and move it to semantic color, but  for basic colors I have seen websites use this syntax.